simpleflow/boto3_utils.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+
+import hashlib
+from contextvars import ContextVar
+from typing import Any
+
+import boto3
+
+from simpleflow.utils import json_dumps
+
+_client_var: ContextVar[dict | None] = ContextVar("boto3_clients", default=None)
+
+
+def get_or_create_boto3_client(*, region_name: str | None, service_name: str, **kwargs: Any):
+    d = {
+        "region_name": region_name,
+        "service_name": service_name,
+    }
+    d.update(kwargs)
+    key = hashlib.sha1(json_dumps(d).encode()).hexdigest()
+    boto3_clients = _client_var.get()
+    if boto3_clients is None:
```

Comment:
and trigger the if on an empty dict
```suggestion
    if not boto3_clients:
```